### PR TITLE
ODBC driver: update and fix AWS SDK with CURL

### PR DESCRIPTION
**Pull request by Yury-Fridlyand**
### Description
- `AWS SDK` compilation workflow changed to use `libcurl` instead of `winhttp` as a transport library
  - `libcurl` shipped by `vcpkg` according to the [doc](https://curl.se/docs/install.html)
- AWS SDK updated
- Fixed crash in driver caused by improper libcurl resource release (incorrect order) (see pic below)

### TODO
- Optimize `AWS SDK` checkout (~90k files)
  - Don't checkout all submodules
  - Don't checkout entire library
- Optimize `vcpkg` deploy to make it faster
  - Checkout `libcurl` only if possible
- Fix compilation of 64 bit driver (`libcurl` in `vcpkg` is 32 only in current config) - see PR #29 
- Don't compile tests (unit tests _aka_ `ut` and google tests)
- Compile static library of driver too
- Fix Power BI connector to set `Certificate validation` flag to false
- Fix installer to include all required files (`libcurl`, `zlib`, etc)
 
### Issues Resolved
https://github.com/opendistro-for-elasticsearch/sql/issues/686
https://github.com/opendistro-for-elasticsearch/sql/issues/783
https://github.com/opensearch-project/sql/issues/302
https://github.com/opensearch-project/sql/issues/280
Applicable for `ODFE` too
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
image

### CODE REVIEW

File: sql-odbc/src/sqlodbc/opensearch_communication.cpp [hunk 1]
```diff
@@ -243,6 +243,7 @@ OpenSearchCommunication::OpenSearchCommunication()
 }
 
 OpenSearchCommunication::~OpenSearchCommunication() {
+    m_http_client.~shared_ptr();
```
[the remark this answers is not in the document]
Comment (Yury-Fridlyand) [reply sql-odbc/src/sqlodbc/opensearch_communication.cpp]: Fixed in b47cf984 and b7140bf0.

File: sql-odbc/scripts/build_aws-sdk-cpp.ps1 [hunk 2]
```diff
@@ -23,13 +25,19 @@ Set-Location $BUILD_DIR
 # Configure and build 
 cmake $SRC_DIR `
     -A $WIN_ARCH `
+	-D CMAKE_VERBOSE_MAKEFILE=ON `
```
[the remark this answers is not in the document]
Comment (Yury-Fridlyand) [reply sql-odbc/scripts/build_aws-sdk-cpp.ps1]: Fixed in 21601044, thank you.

**Review by kylepbit — COMMENTED** [review]
This LGTM, once the 64-bit concerns have been addressed. Is the upgrade of the SDK done across platforms?

File: sql-odbc/scripts/build_aws-sdk-cpp.ps1 [hunk 3]
```diff
@@ -23,13 +25,19 @@ Set-Location $BUILD_DIR
...
     -D CMAKE_INSTALL_PREFIX=$INSTALL_DIR `
     -D CMAKE_BUILD_TYPE=$CONFIGURATION `
     -D BUILD_ONLY="core" `
     -D ENABLE_UNITY_BUILD="ON" `
     -D CUSTOM_MEMORY_MANAGEMENT="OFF" `
     -D ENABLE_RTTI="OFF" `
-    -D ENABLE_TESTING="OFF"
+    -D ENABLE_TESTING="OFF" `
+    -D FORCE_CURL="ON" `
+    -D ENALBE_CURL_CLIENT="ON" `
```
Comment (guiangumpac) [inline sql-odbc/scripts/build_aws-sdk-cpp.ps1:37]: ENABLE is spelt wrong. Does this work?

Comment (Yury-Fridlyand) [reply sql-odbc/scripts/build_aws-sdk-cpp.ps1]: Fixed in e0672b3b. Hopefully, it affected nothing due to `FORCE_CURL`. Maybe it is odd.

**Pull request closed without merging**